docs/api.asciidoc

```diff
@@ -0,0 +1,191 @@
+----
+
+ * `message`: The message as a string.
+ * `param_message`: Alternatively, a parametrized message as a dictionary.
+    The dictionary contains two values: `message`, and `params`.
+    This allows the APM server to group messages together that share the same
+    parametrized message. Example:
++
+[source,python]
+----
+client.capture_message(param_message={
+    'message': 'Billing process for %s succeeded. Amount: %s',
+    'params': (customer.id, order.total_amount),
+})
+----
++
+ * `stack`: If set to `True` (the default), a stacktrace from the call site will be captured.
+ * `exc_info`: A `(type, value, traceback)` tuple as returned by
+   https://docs.python.org/3/library/sys.html#sys.exc_info[`sys.exc_info()`].
+   If not provided, it will be captured automatically, if `capture_message()` was called in an `except` block.
+ * `date`: A `datetime.datetime` object representing the occurrence time of the error.
+   If left empty, it defaults to `datetime.datetime.utcnow()`.
+ * `context`: A dictionary with contextual information. This dictionary must follow the 
+    https://www.elastic.co/guide/en/apm/server/current/error-api.html#error-context-schema[Context] schema definition.
+ * `custom`: A dictionary of custom data you want to attach to the event.
+
+Returns the id of the message as a string.
+
+NOTE: either the `message` or the `param_message` argument is required.
+
+[float]
+[[transaction-api]]
+==== Transactions
+
+[float]
+[[client-api-begin-transaction]]
+===== `Client.begin_transaction()`
+Begin tracking a transaction.
+Should be called e.g. at the beginning of a request or when starting a background task. Example:
+
+[source,python]
+----
+client.begin_transaction('processors')
+----
+
+ * `transaction_type`: A string describing the type of the transaction, e.g. `'request'` or `'celery'`.
+
+[float]
+[[client-api-end-transaction]]
+===== `Client.end_transaction()`
+End tracking the transaction.
+Should be called e.g. at the end of a request or when ending a background task. Example:
+
+[source,python]
+----
+client.end_transaction('myapp.billing_process', processor.status)
+----
+
+ * `name`: A string describing the name of the transaction, e.g. `process_order`.
+   This is typically the name of the view/controller that handles the request, or the route name.
+ * `result`: A string describing the result of the transaction.
+   This is typicaly the HTTP status code, or e.g. `'success'` for a background task.
+
+
+[float]
+[[api-other]]
+=== Other APIs
+
+[float]
+[[api-elasticapm-instrument]]
+==== `elasticapm.instrument()`
+
+Instruments libraries automatically.
+This includes a wide range of standard library and 3rd party modules.
+A list of instrumented modules can be found in `elasticapm.intrumentation.register`.
+This function should be called as early as possibly in the startup of your application.
+For supported frameworks, this is called automatically. Example:
```

Comment:
A link to the docs listing the supported framework would be a nice to have here.